src/dune_rules/module.ml

```diff
@@ -377,7 +377,10 @@ let decode_old ~src_dir =
        if exists then
          let module_basename n ~(ml_kind : Ml_kind.t) ~(dialect : Dialect.t) =
            let n = Module_name.to_string n in
-           String.lowercase n ^ Dialect.extension dialect ml_kind
+           let ext =
+             Option.value (Dialect.extension dialect ml_kind) ~default:""
```

Comment:
I don't think that this case is correct - it should be an error to refer to the filename of an interface for a dialect that has no interfaces.

Reply:
done